syntax = "proto3";

service digitalCashService {
    rpc sendToBankFromCustomer(Message) returns (Message);
	rpc sendToMerchantFromCustomer(Message) returns (ack);
	rpc sendToBankFromMerchant(Message) returns (Message);
	rpc sendToCustomerFromMerchant(Message) returns (Message);
	rpc ping(pingMessage) returns (ack);
}

message Message {
	string messageData = 1;
	int32 numberOfMoneyOrders = 2;
	string MOString = 3;
}

message ack {
	bool success = 1;
	string message = 2;
}

message pingMessage {
	string message = 1;
}